data/data/powervs/cluster/power_network/pi_network.tf

```
@@ -20,7 +20,7 @@ resource "ibm_pi_dhcp" "new_dhcp_service" {
 }
 
 resource "ibm_pi_cloud_connection" "new_cloud_connection" {
-  count                              = var.cloud_conn_name == "" ? 1 : 0
+  count                              = var.transit_gateway_enabled ? 0 : (var.cloud_conn_name == "" ? 1 : 0)
```

Nit: maybe this looks a bit better?
```suggestion
  count                              = (var.transit_gateway_enabled || var.cloud_conn_name != "") ? 0 : 1
```

---

I hope this does not introduce a regression.